c10/core/TensorOptions.h

```diff
@@ -357,8 +357,18 @@ struct C10_API TensorOptions {
       case Layout::Strided:
         switch (device().type()) {
           case DeviceType::CPU:
+            // This is a hack to make TensorOptions based dispatch work for
+            // complex backends.
+            // TODO: Find a better solution and extend it to other backends that
+            // share the same Layout and DeviceType
```

Comment:
I don't see what is wrong with this. `computeTensorTypeId` takes the tuple (device type, layout, dtype) and determines what the appropriate dispatch key is. It special cases strided, cpu, etc.; why can't it special case dtype? The only trouble is if you want *open* dtype/layout registration, but this situation wouldn't apply to complex anyway since it's already in our built-in enum.